solutions/baltic-20-VillageMinimum.mdx

```diff
@@ -0,0 +1,123 @@
+---
+id: baltic-20-VillageMinimum
+source: Baltic OI 2020
+title: Village (Minimum)
+author: Chuyang Wang 
+---
+
+[Official Analysis](http://www.boi2020.lv/data/spoilers/village.pdf)
+
+## Explanation
+
+This problem can be solved by a greedy approach. In particular, we perform a
+depth-first search on the given tree. For each node (villager) that is still at
+its original position, we swap it with its only neighbor, namely its parent
+node. For the root node which does not have a parent node, we can swap it with
+any of its children.
+
+By swapping two nodes where at least one of them is not processed yet, we can
+guarantee that no node will be swapped back to its original position. We can
+also show that swapping between neighbors is the optimal solution: Suppose we
+would swap two unprocessed nodes which are not neighbors. The distance the two
+nodes need to travel would then be equal or larger than 4. If we instead just swap them
+with their respective neighbors, the distance they need to travel is exactly 4.
+Therefore, swapping nodes that are not neighbors never lead to a better
+solution.
+
+To illustrate this idea, let's consider the following example: 
+
+![Example 1](baltic-20-VillageMinimum/example1.png)
+
+Nodes 2 and 3 are not swapped yet. Now, if we swap node 3 and 4 directly, the
```

Comment:
i don't see a node 4 in the diagram?

Reply:
opps gonna fix that
<del>changed the diagramm in between and forgot to update the text</del>